lib/fluent/plugin/in_object_space_dump.rb

```diff
@@ -0,0 +1,59 @@
+# Copyright 2019 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
```

Comment:
Can we add some description for what this plugin does, and how we expect it to be used and how to configure it? Something as simple as https://github.com/GoogleCloudPlatform/fluent-plugin-google-cloud/blob/master/README.rdoc.

Do we claim this as something we officially support? I'm leaning towards treating it just as a debugging tool. External users are free to use it, but we do not have commitment / SLO around related issues.

Reply:
The description is in the class comment. what else would be helpful?

It is in our best interest that this works for its intended purpose, but beyond that, I'd rather not document this as officially supported, if it means more work for us. What do you have in mind?

Reply:
Some instruction (whether here or in our internal team site is fine) about how to use it would be helpful, e.g.:

```
To embed insertIds into log entries, specify @type add_insert_ids in a filter clause of your Fluentd configuration file, for example:

<filter **>
  @type add_insert_ids
  insert_id_key my_insert_id_field_name # Optional.
</filter>
insert_id_key can be used to customize the insertId field name.
```

I agree we probably do not want to officially support this.

Reply:
I'm planning to write instructions for profiling memory once I have successfully completed the task. I don't think that instructions on how to initialize the plugin are helpful without the rest. Do you want me to hold off this change until then?